cmd/soci/main.go

```diff
@@ -97,8 +103,67 @@ func main() {
 		commands.RebuildDBCommand,
 	}
 
+	app.Before = func(ctx *cli.Context) error {
+		if rootlessutil.IsRootlessParent() {
+			return parentMain()
+		}
+
+		return nil
+	}
+
 	if err := app.Run(os.Args); err != nil {
 		fmt.Fprintf(os.Stderr, "soci: %v\n", err)
 		os.Exit(1)
 	}
 }
+
+func parentMain() error {
```

Comment:
It looks like a lot of this code came directly from nerdctl. That's fine since it's apache 2.0, but we need to add a comment that indicates where this came from and a note about the license/copywrite. 

If we can pull this out into another file, it's pretty easy to mark it with a header like: https://github.com/awslabs/soci-snapshotter/blob/6dcfca7bc0c70b00a5f23813d468379d6fbc8de5/fs/fs.go#L1-L31